include/mbedtls/threading.h

```diff
@@ -39,16 +39,72 @@ extern "C" {
 #define MBEDTLS_ERR_THREADING_BAD_INPUT_DATA              -0x001C  /**< Bad input parameters to function. */
 #define MBEDTLS_ERR_THREADING_MUTEX_ERROR                 -0x001E  /**< Locking / unlocking / free failed with error code. */
 
+/* Forwards declaration of mutex type  */
+typedef struct _mbedtls_threading_mutex_t mbedtls_threading_mutex_t;
```

Comment:
I presume that this line was added here to avoid the additional preprocessor condition checking for `MBEDTLS_THREADING_ALT` and `MBEDTLS_THREADING_PTHREAD`. However, in my opinion this makes the file a little disorganised, not to mention that this does not compile. I would suggest reorganising the code, perhaps somethins like below:
```
#if defined(MBEDTLS_THREADING_C)
// Definition for the mbedtls_threading_mutex_t struct for each of the cases

// User defined types for function pointers.

// Remaining code for each of the compilation cases.

#endif
```